src/audio/drc/drc_math_lib/asin/c_fixpt/drc_asin.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * File: drc_asin.c
+ *
+ * Author :Shriram Shastry
+ * Copyright (c) 2016, Intel Corporation All rights reserved.
+ */
+
+/* Include Files */
+#include "drc_asin.h"
+
+/* Function Declarations */
+static int asr_s32(int u, unsigned int n);
+static signed char iScalarCordicAsin(signed char sinValue, short nIters, const
+  signed char LUT[6]);
+
+/* Function Definitions */
+
+/*
+ * Arguments    : int u
+ *                unsigned int n
+ * Return Type  : int
+ */
+static int asr_s32(int u, unsigned int n)
```

Comment:
asr isn't clear to everyone, please provide a description. Please also apply to other functions.

Reply:
ASR : Arithmetic shift right  for 32 bit wordlength
    Arithmatic bit right shift is performed for the cordic iteration
    for angles less than pi/4, we initialize (x,y) along the
    x-axis[xShift,xDShift]. 
    For angles greater than or equal to pi/4, we initialize (x,y)
    along the y-axis[yShift,yDShift]. 
    This improves the accuracy of the algorithm near the edge of the domain of convergence.
    Variables j,k are derived from the Cordiac Iteration (WordLength-1)
     j = min(bitsll(i,1),maxShift)  // bitsll logical left shift
     k = min(i - 1,maxShift);

Reply:
@ShriramShastry that's good info above, but this needs to go into the comment above the function implementation as a doxygen comment. This way we can process all the API comments into an online developer guide.